### PR TITLE
Remove unused components

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/16944 & https://github.com/giantswarm/giantswarm/issues/16945

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Start of a new history. Previous commits have been imported from https://github.com/giantswarm/config-controller.
```
Comment (kopiczko) [inline CHANGELOG.md:12]: I'd say it's ok to wipe out all the other changelogs.

File: cmd/generate/runner.go [hunk 2]
```diff
@@ -66,14 +65,11 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
...
 	in := generator.GenerateInput{
-		App:           r.flag.App,
-		ConfigVersion: r.flag.ConfigVersion,
-
+		App:       r.flag.App,
 		Name:      r.flag.Name,
 		Namespace: r.flag.Namespace,
 
 		ExtraAnnotations: map[string]string{
-			meta.Annotation.ConfigVersion.Key():   r.flag.ConfigVersion,
```
Comment (kopiczko) [inline cmd/generate/runner.go:76]: Don't we want to set the output of `git describe --tags` here?

File: flag/service/service.go [hunk 3]
```diff
@@ -4,16 +4,12 @@ import (
 	"github.com/giantswarm/operatorkit/v4/pkg/flag/service/kubernetes"
```
Comment (kopiczko) [inline flag/service/service.go:4]: Do we need the `flag` package at all?

File: main.go [hunk 4]
```diff
@@ -133,16 +88,13 @@ func mainE(ctx context.Context) error {
...
 	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.Address, "http://127.0.0.1:6443", "Address used to connect to Kubernetes. When empty in-cluster config is created.")
 	daemonCommand.PersistentFlags().Bool(f.Service.Kubernetes.InCluster, false, "Whether to use the in-cluster config to authenticate with Kubernetes.")
 	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.KubeConfig, "", "KubeConfig used to connect to Kubernetes. When empty other settings are used.")
 	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.CAFile, "", "Certificate authority file path to use to authenticate with Kubernetes.")
 	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.CrtFile, "", "Certificate file path to use to authenticate with Kubernetes.")
 	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.KeyFile, "", "Key file path to use to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Vault.Address, "", "Vault server address")
-	daemonCommand.PersistentFlags().String(f.Service.Vault.Token, "", "Vault server address")
```
Comment (kopiczko) [inline main.go:145]: I think we don't need the whole `daemonCommand `. We are not going to run it asa controller anymore.

File: pkg/filesystem/filesystem.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"io/ioutil"
+	"os"
+)
+
+type Store struct{}
+
+func (s *Store) ReadFile(path string) ([]byte, error) {
+	return os.ReadFile(path)
+
```
Comment (kopiczko) [inline pkg/filesystem/filesystem.go:12]:
```suggestion
```

File: cmd/generate/runner.go (hunk 2 above)
Comment (kubasobon) [reply cmd/generate/runner.go:76]: This change will go here: https://github.com/giantswarm/konfigure/pull/3

File: main.go [hunk 6]
```diff
@@ -130,23 +80,6 @@ func mainE(ctx context.Context) error {
...
-	daemonCommand.PersistentFlags().Bool(f.Service.Kubernetes.InCluster, false, "Whether to use the in-cluster config to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.KubeConfig, "", "KubeConfig used to connect to Kubernetes. When empty other settings are used.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.CAFile, "", "Certificate authority file path to use to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.CrtFile, "", "Certificate file path to use to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.KeyFile, "", "Key file path to use to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Vault.Address, "", "Vault server address")
-	daemonCommand.PersistentFlags().String(f.Service.Vault.Token, "", "Vault server address")
-
-	newCommand.CobraCommand().SilenceErrors = true
-	newCommand.CobraCommand().SilenceUsage = true
```
Comment (kopiczko) [inline main.go:148]: This should stay.

**Review by kopiczko — APPROVED** [review]
Sort the CI and I think we are good here. There are 2 lines to restore.

**Pull request merged**